src/Http/Http/src/Timeouts/RequestTimeoutPolicy.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http.Timeouts;
+
+/// <summary>
+/// Defines the policy for request timeouts middleware.
+/// </summary>
+public sealed class RequestTimeoutPolicy
+{
+    /// <summary>
+    /// The timeout to apply.
+    /// </summary>
+    public TimeSpan? Timeout { get; init; }
+
+    /// <summary>
+    /// Status code to be set in response when a timeout results in an <see cref="OperationCanceledException" /> being caught by the middleware.
+    /// The status code cannot be applied if the response has already started.
+    /// 504 will be used if none is specified.
+    /// </summary>
+    public int? TimeoutStatusCode { get; init; }
```

Comment:
Wouldn't it better to use `HttpStatusCode` enum instead of `int` here?

Reply:
From https://github.com/aspnet/Mvc/pull/7659#issuecomment-386726928

> The System.Net.HttpStatusCode is an older .NET enum implemented for HTTP client scenarios that has a number of design issues that we decided to not to propagate in ASP.NET Core. Instead ASP.NET Core has it's own [HTTP abstractions](https://github.com/aspnet/httpabstractions) including it's own [StatusCodes](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.statuscodes) type.